client/src/components/TaskDurationBarChart/TaskDurationBarChart.js

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import styles from "./TaskDurationBarChart.module.css";
+import PropTypes from "prop-types";
+import {
+  BarChart,
+  Bar,
+  XAxis,
+  YAxis,
+  ReferenceLine,
+  ResponsiveContainer,
+} from "recharts";
+
+export const TaskDurationBarChart = ({ taskResponseData }) => {
+  const taskDurationDifferences = [];
+
+  console.log("hi");
+  console.log(taskResponseData);
+  for (const task of taskResponseData) {
+    const taskDifference = {
+      taskName: task.title,
+      percentDifference: 0,
+    };
+
+    console.log(task);
+    const plannedDuration =
+      task.originalEndDate.getTime() - task.originalStartDate.getTime();
+    const actualDuration = task.endDate.getTime() - task.startDate.getTime();
+    taskDifference.percentDifference =
+      (actualDuration - plannedDuration) / plannedDuration;
+
+    if (task.taskStatus === "DONE")
+      taskDurationDifferences.push(taskDifference);
+  }
+
+  return (
+    <ResponsiveContainer width="100%" height="100%">
+      <div className={styles.scrollableTaskChart}>
+        <BarChart
+          width={500}
+          height={300}
+          data={taskDurationDifferences}
+          margin={{
+            top: 5,
+            right: 30,
+            left: 20,
+            bottom: 5,
+          }}
+        >
+          <XAxis dataKey="taskName" stroke="black" />
+          <YAxis
+            stroke="black"
+            label={{
+              value: "Difference (%) in Time Spent",
```

Comment:
Is there a way that it's not done in percentages? (Perhaps showing the differences in minutes might be more effective?)
![image](https://user-images.githubusercontent.com/24628243/162017763-c92252e4-d89c-4a9b-90e5-ad94d014d152.png)

Reply:
Justification for Percentage Difference:
Spending an extra 10 minutes on a 3 hour long task is much less significant compared to spending extra 10 minutes on a 10 minute long task.

Reply:
Agreed!